src/payment/strategies/stripev3/stripev3-payment-strategy.ts

```diff
@@ -184,13 +184,20 @@ export default class StripeV3PaymentStrategy implements PaymentStrategy {
             try {
                 return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
             } catch (error) {
+                if (this._isCancellationError(result.error)) {
+                    throw new PaymentMethodCancelledError();
+                }
                 throw this._handleEmptyPaymentIntentError(error, result.error);
             }
         }
 
         throw error;
     }
 
+    private _isCancellationError(stripeError: StripeError | undefined) {
+        return stripeError && (stripeError.payment_intent?.last_payment_error?.message?.indexOf('canceled') !== -1);
```

Comment:
I'm not sure about using indexOf, is `cancelled` the full description?

Reply:
Hi @vmparra, The full error message that I get when the user closes the auth modal is this: 

_"The latest payment attempt of this PaymentIntent has failed or been canceled, and the attached payment method has been removed. You can try confirming again with a new payment method."_

And the message that I get when authentication fails is: 

_"The provided PaymentMethod has failed authentication. You can provide payment_method_data or a new PaymentMethod to attempt to fulfill this PaymentIntent again."_

To differentiate between one error and the other, I based myself on the fact that in the cancellation the message contains the word “canceled”.

Reply:
Great, tanks for the explanation.

Reply:
Yes, we get the error code `payment_intent_authentication_failure` the problem is that it is the same error code for both errors when authentication fails and when the user closes the auth modal.

Reply:
This does feel pretty fragile, do the scenarios produce errors with different cancellation reasons?

https://stripe.com/docs/api/errors#errors-payment_intent-cancellation_reason

Reply:
`cancellation_reason` is null. The only thing that differentiates the errors is the error message.

<img width="645" alt="Screen Shot 2021-07-30 at 5 33 11 PM" src="https://user-images.githubusercontent.com/69487174/127718801-719a3f79-5616-476f-b1f0-723e5662ab2b.png">